### PR TITLE
Improve linting process

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

- Created a github action workflow called "CI" which will run on opened PRs
- Run ESLint from this workflow
- Removing ESLint from the pre-commit phase

### Potential Breaking Changes

None, ci stuff only

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: .github/actions/lint/lintChangedFiles.mjs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+const changedFilesWithoutDuplicates = [...new Set(changedFiles)];
+
+const ALLOWED_EXTENSIONS = ['js', 'ts', 'jsx', 'tsx'];
+
+(async function main() {
+    const eslint = new ESLint();
+
+    console.log('\x1b[32mList of changed files', changedFilesWithoutDuplicates);
+
+    // // remove all files that are not an allowed extension from the array
```
Comment (aiheon) [inline .github/actions/lint/lintChangedFiles.mjs:14]:
```suggestion
    // remove all files that are not an allowed extension from the array
```

File: .github/actions/setup/action.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        - uses: pnpm/action-setup@35ab4267a1a21c8e8cb1c087cf1642e891ff57bd
+          with:
+              version: 7
+              run_install: false
+        - name: Get pnpm store directory
+          id: pnpm-cache
+          shell: bash
+          run: |
+              echo "::set-output name=pnpm_cache_dir::$(pnpm store path)"
+        - uses: actions/cache@v3
```
Comment (aiheon) [inline .github/actions/setup/action.yml:16]: we can because plasma is open source? 😁  Noice!

Comment (gdostie) [reply .github/actions/setup/action.yml:16]: Yup, we can also use the github hosted runners so it costs us nothing to run

**Review by aiheon — APPROVED** [review]

File: .github/actions/setup/action.yml (hunk 2 above)
Comment (aiheon) [reply .github/actions/setup/action.yml:16]: that's cool 😎

**Review by toofff — APPROVED** [review]
LGTM

**Review by GermainBergeron — APPROVED** [review]

**Pull request merged**